ledger/complete/mtrie/node/node.go

```diff
@@ -122,12 +122,91 @@ func NewInterimNode(height int, lchild, rchild *Node) *Node {
 	return n
 }
 
+// CopyAndPromoteLeafNode makes a copy of a node and moves it one level higher to replace
+// the parent node, this method should only be called for leaf nodes
+// depending on where this node is located to the parent the
+// hash value would be different, if isLeft is set to true the original place
+// of the node n was left child of its parent so the hash value would be adjusted accordingly
+func (n *Node) copyAndPromoteLeafNode(isLeft bool) *Node {
+	// note path is an arrays (not slice) so it would be coppied
+	newNode := &Node{
+		height:   n.height + 1,
+		path:     n.path,
+		payload:  n.payload.DeepCopy(),
```

Comment:
not sure if we need this deep copy or not :-?

Reply:
I think we do since the input is a pointer and we don't want the payload data to be updated. This is what `NewLeaf` also does.